pkg/common/cns-lib/volume/manager.go

```diff
@@ -279,10 +279,21 @@ func (m *defaultManager) CreateVolume(ctx context.Context, spec *cnstypes.CnsVol
 		datastoreURL = dsMo.Summary.Url
 	}
 
-	blockBackingDetails, ok := spec.BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails)
+	blockBackingDetails, isBlockBackingDetails := spec.BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails)
```

Comment:
As discussed Can you move the creation of blockBackingDetails and  fileBackingDetails objects inside the if condition after we identify block or file using `spec.VolumeType == string(cnstypes.CnsVolumeTypeBlock)` and `spec.VolumeType == string(cnstypes.CnsVolumeTypeFile)`.